docs/developer/design/authentication.rst

```diff
@@ -0,0 +1,154 @@
+Authentication
+==============
+
+Todo
+----
+
+- Provide uml diagrams for Dex + MetalK8s authentication flow
+
+Open questions
+--------------
+
+- Is support for Dex monitoring and alerting on Grafana
+  something we should think about now?
+
+Abstract
+--------
+
+To be able to identify who a user is and to manage(Create/Edit/Delete)
+these user identities, MetalK8s needs a middleman that acts in this light
+as an Identity Provider.
+
+To accomplish this, we introduce the concept of an OpenID Connect identity
+provider which has as role identity federation in a typical MetalK8s cluster.
+
+
+Scope
+-----
+
+The scope of **Authentication** will be split into 3 main iterations as
+detailed in the implementation section of this document.
+
+Requirements
+------------
+
+User Requirements for Authentication in MetalK8s has been discussed and is
+well documented. For more information, refer to -> [link to Requirements DOC]
+
+
+Design
+------
+
+For authentication in **MetalK8s**, we are going to use Dex, Dex
+connectors/plugins and it's integrated support for other
+External IDP Integration as an Identity Provider.
```

Comment:
Not super clear nor sufficient. We will mostly use Dex as an **OIDC Provider**, i.e. some service that can issue OIDC tokens for a user to authenticate against a third-party application (here, K8s API). As a bonus, Dex also can be used as an actual **Identity Provider**, in which case it acts as both parties in the OIDC flow.

To me, the design should first state that we **want** to use OIDC as the authentication mechanism against K8s API for human users, and then explain that we chose Dex for its ability to connect non-OIDC compatible IdPs, such as Active Directory, in that OIDC flow. And in the details of why we chose Dex, add a mention that we can still use it as a fallback IdP, in case the external ones are not reachable/configured.